app/src/main/java/com/ml/shubham0204/docqa/ui/screens/docs/DocsViewModel.kt

```diff
@@ -55,6 +59,38 @@ class DocsViewModel(
         }
     }
 
+    suspend fun addDocumentFromUrl(url: String, context: Context, callback: (Boolean) -> Unit) = withContext(Dispatchers.IO){
```

Comment:
Instead of naming the variable as `callback`, this lambda function can be better named as `onDownloadComplete` which precisely tells when the lambda function will be called.